spec/unit/provider/systemd_unit_spec.rb

```diff
@@ -222,6 +222,10 @@
 
     it "loads the user unit content if the file exists and user is set" do
       new_resource.user("joe")
+      allow(Chef::Resource::SystemdUnit).to receive(:new)
+        .with(unit_name)
+        .and_return(new_resource)
```

Comment:
this looks incorrect, we're testing load_current_resource so should be calling this directly to get a blank object to fill out, by cloning the new_resource this is almost certainly masking over bugs rather than fixing them